msal/build.gradle

```diff
@@ -124,6 +134,38 @@ task sourcesJar(type: Jar) {
     destinationDir = reporting.file("$project.buildDir/outputs/jar/")
 }
 
+task createMsalCoverageReportTask() {
+    group = "Msal Reporting"
+    description = "Generate Coverage Report tasks for every build Flavour"
+    android.libraryVariants.all { variant ->
+        def buildVariant = variant.name
+        def testTaskName = "test${buildVariant.capitalize()}UnitTest"
+
+        // Tasks for generating Coverage reports for every buildVariant
+        task "${buildVariant}MsalCoverageReport" (type:JacocoReport, dependsOn: ["$testTaskName", "create${buildVariant.capitalize()}CoverageReport"]) {
+            group = "Msal Reporting"
+            description = "Generate Msal Coverage Reports(UnitTest & AndroidTests) on the ${buildVariant.capitalize()}"
+
+            def coverageSourceDirs = "${projectDir}/src/main/java"
```

Comment:
How is the source code in common dependency being considered? It did not seem to be considered (at least when running from Android-Complete)

Reply:
No that is not the case, the actual path of coverageSourceDirs is "android-complete\msal\msal/src/main/java" , where path of projectDir is "android-complete\msal\msal". So jacoco use all the src classes and instrument it to generate coverage report. so that is the purpose of this coverageSourceDirs .

Reply:
common repos are included in coverage report